_topics/en/compact-block-filters.md

```diff
@@ -0,0 +1,76 @@
+---
+title: Compact block filters
+
+## Optional.  An entry will be added to the topics index for each alias
+aliases:
+  - BIP157
+  - BIP158
+  - Neutrino protocol
+
+## Required.  At least one category to which this topic belongs.  See
+## schema for options
+categories:
+  - Privacy Enhancements
+  - Lightweight Client Support
+  - P2P Network Protocol
+
+## Required.  Use Markdown formatting.  Only one paragraph.  No links allowed.
+excerpt: >
+  **Compact block filters** are a condensed representation of the
+  contents of a block that allow wallets to determine whether the block
+  contains any transactions affecting the wallet.
+
+## Optional.  Use Markdown formatting.  Multiple paragraphs.  Links allowed.
+extended_summary: |
+  A full node with a copy of a block and the data necessary to validate
+  that block uses [BIP158][] to create a Golomb-Rice Coded Set (GCS) of
+  data from that block (such as the prevout script from each input spent
```

Comment:
What do you mean by "prevout scipt"? I haven't come across that terminology before.

Reply:
prevout = previous output; script = scriptPubKey.  So the encumbrance that the input's scriptSig or witness fulfills.  Grepping the Bitcoin Core source, I find four instances of "prevout scriptPubKey" (all in the tests), so maybe I should change it to say that?

Reply:
I think 'scriptPubKey of every output spent in the block and scriptPubKey of every new output created in the block' (or something similar) is clearer.

Reply:
I think "scriptPubKey of every output spent in the block" is ambiguous.  It could refer (correctly) to the prevout scriptPubKeys or it could refer to a subset of that being just the scriptPubKeys of transactions in the block that were spent by later transactions also included in the same block.  I think I'll just remove the whole parenthetical; anyone who wants to know precisely what data is included can read BIP158 or our other links.

Reply:
Um, since the earliest version of that documentation, there has been a todo:

```
<!-- SOMEDAY: we need more terms than just output/input to denote the
various ways the outputs/inputs are used, such as "prevout", "nextout",
"curout", "curin", "nextin".  (Is there any use for "previn"?)  Someday,
when I'm terribly bored, I should rewrite this whole transaction section
to use those terms and then get feedback to see if it actually helps. -harding -->
```

You can consider my subsequent adoption of the term `prevout` in later documentation as work towards solving that terminology problem I first encountered in early 2014.  I'm hardly alone in that, the term is being used in many other pieces of technical documentation, including BIP68 sequence numbers, BIP143 segwit v0 signature algorithm, BIP174 PSBTs, bip-taproot, and (perhaps most obviously) bip-anyprevout.

In this case, it doesn't matter---I'm just deleting this entire parenthetical---but we'd have to have a longer conversation before you get me to give up on the term altogether as I think it's an convenient way to express a frequently-used (and particularly-import) part of validation data.

Reply:
For future reference, `prevout` in those documents refers to the _outpoint_ (the hash and txOut index of the previous transaction output), and not the transaction output itself.